lib/spec_file_generator.js

```diff
@@ -43,16 +43,19 @@ function dependenciesToRequires(deps) {
   for (var i = 0; i < sortedDeps.length; i++) {
     var dep = sortedDeps[i];
     var version = deps[dep];
-    if (version[0] === '^' || version[0] === '~') {
-      var min_version = version.substr(1);
-      dependencies.push('npm(' + dep + ') >= ' + min_version);
-
-      var max_version = semver.inc(min_version, version[0] === '^' ? 'major' : 'minor');
-      dependencies.push('npm(' + dep + ') < ' + max_version);
-    } else if (isNaN(parseInt(version[0], 10))) {
-        dependencies.push('npm(' + dep + ') ' + version);
+    var versionRange = semver.validRange(version);
+
+    // '*' has no version restriction, || can't be expressed with simple deps,
+    // and null means something went wrong. For any of these just output a name
+    // with no version
+    if (versionRange === null || versionRange === '*' || versionRange.includes('||')) {
+      dependencies.push('npm(' + dep + ')');
     } else {
-      dependencies.push('npm(' + dep + ') = ' + version);
+      // we should now have a space-separated list of comparators and version numbers
+      // that can be added to a spec file as-is
+      versionRange.split(' ').forEach((constraint) => {
```

Comment:
How does this handle `>= 1.0 < 2.0`? Wouldn't it end up as 4 entries?

Reply:
Granted, this is dependent on the behavior of semver. At least with the current version, the string that validRange spits out is '>=1.0 <2.0', so there's only one space in there. Tests would help to at least ensure that it keeps working.

What I thought would be nice is to have a library that can parse a semver string into an array of objects describing the constraints, but I figured this would be good for now as a quick fix.

Reply:
I'd be OK with it when there are tests